actionpack/lib/action_dispatch/http/mime_negotiation.rb

```diff
@@ -151,6 +154,14 @@ def valid_accept_header
       def use_accept_header
         !self.class.ignore_accept_header
       end
+
+      private
+        def extract_variant!
+          if parameters.key?(:format)
+            parameters[:format], *variants = parameters[:format].to_s.split("+")
```

Comment:
Also `extract_variant!`, like other rails `exract_...` patterns, implies side effect on the argument, but doesn't imply side effect on setting `self.variant`. Perhaps name this `set_variant`. On same note, could refactor existing `format=` and `formats=` methods to call shared `set_formats` which extracts common logic between the two. (e.g. normalize as `Array(formats)`)